pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,341 @@
+        :return: pydantic Annotation Model of annotations of current project
+        """
+        admin_tuple = tuple_converter(admin)
+        sql_q = f"""
+                select 
+                    {NAMESPACE_COL},
+                    {NAME_COL},
+                    {TAG_COL},
+                    {PRIVATE_COL},
+                    {PROJ_COL}->>'description',
+                    {N_SAMPLES_COL},
+                    {SUBMISSION_DATE_COL},
+                    {LAST_UPDATE_DATE_COL},
+                    {DIGEST_COL}
+                        from {DB_TABLE_NAME}
+                """
+
+        if tag is None:
+            tag = DEFAULT_TAG
+
+        if name:
+            sql_q = f""" {sql_q} where {NAME_COL}=%s and {NAMESPACE_COL}=%s and {TAG_COL}=%s 
+                                and ({PRIVATE_COL} is %s or {NAMESPACE_COL} in %s );"""
+            found_prj = self.con.run_sql_fetchone(
+                sql_q, name, namespace, tag, False, admin_tuple
+            )
+
+        else:
+            _LOGGER.error(
+                "You haven't provided name, digest nor id! Execution is unsuccessful"
+            )
+            _LOGGER.info("Files haven't been downloaded, returning empty dict")
+            return AnnotationModel()
+
+        _LOGGER.info(f"Project has been found!")
+        if len(found_prj) > 0:
+            annot = AnnotationModel(
+                namespace=found_prj[0],
+                name=found_prj[1],
+                tag=found_prj[2],
+                is_private=found_prj[3],
+                description=found_prj[4],
+                number_of_samples=found_prj[5],
+                submission_date=str(found_prj[6]),
+                last_update_date=str(found_prj[7]),
+                digest=found_prj[8],
+            )
+            return annot
+        else:
+            _LOGGER.error(f"Project '{namespace}/{name}:{tag}' was not found.")
+
+        return None
+
+    def _count_projects(
+        self,
+        namespace: str,
+        search_str: str = "",
+        admin: Union[str, List[str]] = False,
+    ) -> int:
+        """
+        Get total number of found projects. [This function is related to _find_projects]
+        :param namespace: namespace where to search for a project
+        :param search_str: search string. will be searched in name, tag and description information
+        :param admin: True, if user is admin for this namespace
+        :return: number of found project in specified namespace
+        """
+        search_str = f"%%{search_str}%%"
+        admin_tuple = tuple_converter(admin)
+        if namespace:
+            and_namespace_sql = f"""AND {NAMESPACE_COL} = %s"""
+            namespace = (namespace,)
+        else:
+            and_namespace_sql = ""
+            namespace = tuple()
+
+        count_sql = f"""
+        select count(*)
+            from {DB_TABLE_NAME} 
+                where ({NAME_COL} ILIKE %s or ({PROJ_COL}->>'description') ILIKE %s or {TAG_COL} ILIKE %s)
```

Comment:
Instead of having the `search_str` default to '', you should actually use a different sql query here, because `LIKE` is super slow.

Reply:
So to clarify, you should only use LIKE if the user is actually requesting a serach. Set the `search_str` default to `None` and module the sql statement you use based on that value.

Reply:
If you want, you can use `EXPLAIN` to make sure I'm right about this before going to the effort to make the changes...

Reply:
Thanks for mentioning it. I didn't know about it, and probably we couldn't even observe performance problems, while testing on small dataset. I agree with all your comments, I will change it